### ISSUE TITLE
[BUG] (type-safe-api) generated hooks don't compile

### ISSUE THREAD

**Issue by cogwirrel** [aws/aws-pdk#604]
#### React Query v5 Released

React Query v5 was released last night, which introduces some breaking changes: https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5

#### Mismatching parameter names

The following Smithy model generates hooks code which doesn't compile, since the name used in the `@httpQuery("myInToken")` trait is the actual name used in the generated clients/handlers (since this in the generated OpenAPI spec), but the value from the paginated trait is used in the generated hooks which points to the Smithy member name.

```smithy
@readonly
@http(method: "GET", uri: "/my-list-operation")
@handler(language: "typescript")
@paginated(inputToken: "inToken", outputToken: "outToken", items: "messages")
operation MyListOperation {
    input := {
        @httpQuery("myInToken")
        inToken: String
    }
    output := {
        @required
        messages: Messages

        outToken: String
    }
}

list Messages {
    member: String
}
```

### Expected Behavior

Code to compile

### Current Behavior

```
api-typescript-react-query-hooks: src/apis/DefaultApiClientProvider.tsx(39,42): error TS2322: Type '{ children: Element; client: QueryClient; context: Context<QueryClient>; }' is not assignable to type 'IntrinsicAttributes & QueryClientProviderProps'.
api-typescript-react-query-hooks:   Property 'context' does not exist on type 'IntrinsicAttributes & QueryClientProviderProps'.
api-typescript-react-query-hooks: src/apis/DefaultApiHooks.ts(54,96): error TS2345: Argument of type '{ inToken: any; name: string; myInToken?: string; }' is not assignable to parameter of type 'SayHelloRequest'.
api-typescript-react-query-hooks:   Object literal may only specify known properties, but 'inToken' does not exist in type 'SayHelloRequest'. Did you mean to write 'myInToken'?
api-typescript-react-query-hooks: src/apis/DefaultApiHooks.ts(54,119): error TS2554: Expected 1-2 arguments, but got 3.
```

### Reproduction Steps

Use Smithy operation above and enable the hooks library.

### Possible Solution

Either validate away this case or map the values in the rendered `x-paginated` extension to point to the OpenAPI name.

### PDK version used

0.22.13

### What languages are you seeing this issue on?

Typescript, Java, Python

### Environment details (OS name and version, etc.)

OSX

**Comment by caopengau** [comment]
It happens during the walkthrough in doc https://aws.github.io/aws-pdk/getting_started/your_first_aws_pdk_project.html
```
myapi-typescript-react-query-hooks: 👾 build » compile | tsc --build
myapi-typescript-react-query-hooks: src/apis/DefaultApiClientProvider.tsx(39,42): error TS2322: Type '{ children: Element; client: QueryClient; context: Context<QueryClient>; }' is not assignable to type 'IntrinsicAttributes & QueryClientProviderProps'.
myapi-typescript-react-query-hooks:   Property 'context' does not exist on type 'IntrinsicAttributes & QueryClientProviderProps'.
myapi-typescript-react-query-hooks: src/apis/DefaultApiHooks.ts(54,71): error TS2554: Expected 1-2 arguments, but got 3.
myapi-typescript-react-query-hooks: 👾 Task "build » compile" failed when executing "tsc --build" (cwd: /Users/pengcao/workspace/my-project/packages/api/generated/libraries/typescript-react-query-hooks)
```

**Comment by cogwirrel** [comment]
Thanks! It looks like the error here is due to some breaking changes in a new version of React query that was released last night: https://tanstack.com/blog/announcing-tanstack-query-v5

I'm working on a fix for both of these issues and will raise a PR shortly! :)

**Comment by cogwirrel** [comment]
For now, as a workaround you can add the following to your `.projenrc.ts` after creating the `TypeSafeApiProject`:

```ts
api.library.typescriptReactQueryHooks!.addDeps("@tanstack/react-query@^4");
```

**Comment by cogwirrel** [comment]
The fix has been released in version `0.22.17`! You can upgrade an existing project with the command:

```
npx npm-check-updates -u --deep @aws/pdk
```

You'll then need to re-synth before building again:

```
pdk
pdk build
```
